proposals/0295-codable-synthesis-for-enums-with-associated-values.md

```diff
@@ -152,8 +151,54 @@ enum StoreCodingKeys: CodingKey {
 }
 ```
 
-Since cases with unlabeled parameters encode into unkeyed containers,
-no `CodingKeys` enum will be generated for them.
+The `encode(to:)` implementation would looks as follows:
+
+```swift
+public func encode(to encoder: Encoder) throws {
+  var container = encoder.container(keyedBy: CodingKeys.self)
+  switch self {
+  case let .load(key):
+    var nestedContainer = container.nestedContainer(keyedBy: LoadCodingKeys.self, forKey: .load)
+    try nestedContainer.encode(key, forKey: .key)
+  case let .store(key, value):
+    var nestedContainer = container.nestedContainer(keyedBy: StoreCodingKeys.self, forKey: .store)
+    try nestedContainer.encode(key, forKey: .key)
+    try nestedContainer.encode(value, forKey: .value)
+  }
+}
+```
+
+and `init(from decoder: Decoder)`
+
+```swift
+public init(from decoder: Decoder) throws {
+  let container = try decoder.container(keyedBy: CodingKeys.self)
+  if container.allKeys.count != 1 {
+    let context = DecodingError.Context(
+      codingPath: container.codingPath,
+      debugDescription: "Invalid number of keys found, expected one.")
+    throw DecodingError.typeMismatch(Command.self, context)
+  }
```

Comment:
Synthesized code for structs seems to ignore any extra keys. Can you do the same for enum cases?

```json
{
  "extra_key": "extra_value",
  "load": {
    "key": "MyKey"
  }
}
```

```swift
public init(from decoder: Decoder) throws {
  let container = try decoder.container(keyedBy: CodingKeys.self)
  let allKeys = container.allKeys
  if allKeys.contains(.load) {
    /* ... */
  } else if allKeys.contains(.store) {
    /* ... */
  } else {
    let context = /* ... */
    throw DecodingError.typeMismatch(Command.self, context)
  }
}
```

Reply:
I don't think that would be a good idea. If the container contains more than one key, it is malformed. Checking for every possible key in there would be very bad for performance.